### DISCUSSION TITLE
Strains and Stresses

### DISCUSSION THREAD

**Discussion by marinerhemant** [BlueQuartzSoftware/DREAM3DNX-Issues#7] [Ideas]
I'm working on manipulating diffraction microscopy data (point clouds and EBSD-like maps) using DREAM3DNX. Does a formulation exist in DREAM3DNX to handle strains and stresses corresponding to vertices? Here are the capabilities I'm looking for:

- [ ] Import strains in matrix notation.
- [ ] Rotation of strains using rotation matrices. Right now, I can only find filters for the rotation of Euler angles or geometry for vertices.
- [ ] Calculation of stresses from strains and stiffness matrix.

If there is interest, I can help in developing filters for these. We have other codes in C, Matlab or Python for doing these operations, we could port them to DREAM3DNX.

**Comment by imikejackson** [comment]
@marinerhemant DREAM3DNX should be able to import the data onto a vertex geometry.  You would need the XYZ coordinate of each vertex and an associated strain for that vertex. Unfortunately at this point we are stuck. DREAM3D at this point does not have a general "matrix" operations type of filter. We would need something similar, as you stated, to "Rotate Euler Angles" or "Apply Transformation to Geometry".

I don't believe (in any version of DREAM3D) that we have the calculation of Stresses from strains and stiffness matrix.


We would definitely be interested in having these kinds of codes inside of DREAM3D-NX. The main repository for the filters is https://www.github.com/bluequartzsoftware/simplnx. The DREAM3D-NX repository is currently private by invitation only. When it comes to writing the filters there are a few choices to you. Using C++ is the most documented and has the most examples. We are now generating anaconda python packages of 'simplnx' which is the low level python bindings for writing filters. You could go that way but those filters would _only_ be available in the Anaconda package distribution of DREAM3D-NX.

If ANL is going to be a regular user of DREAM3D-NX I can speak with our funding agent to attempt to get some support to port your filters. We can show you how to do it, or we can do the ports with some help from ANL in the form of example data sets, some final code review and documentation review.

**Reply by marinerhemant** [reply]
@imikejackson this is great news. 

Yes, we are planning to be regular users of DREAM3D-NX for multiple techniques: EBSD, HEDM (FF,NF,PF), BCDI, Laue Microdiffraction etc. We can talk more about what is the best path forward: please contact me at hsharma@anl.gov, I could not find contact info for you.
